### DISCUSSION TITLE
Upgrade from pyright 1.1.344 to 1.1.357 gave these new errors

### DISCUSSION THREAD

**Discussion by GhostlyCipher** [microsoft/pyright#7635] [Q&A] [answered]
I have this project where I use pyspark and after a recent update in pyright from 1.1.344 to 1.1.357, I've started getting this error.
I think the code is self explanatory. There is a variable `repartition_columns` which is a list of strings. I pass that variable to the repartition function. It works exactly like I expect it to but pyright is raising an error.
Not sure if I'm doing something wrong or if it's a bug.

```
repartition_columns = table.repartition_keys
df.repartition(*repartition_columns)
```

Datatypes used here
```
(variable) df: pyspark.sql.DataFrame
(variable) repart_cols: List[str] | None
```

Function definition of repartition. This is an overloaded function. Providing both here:
```
@overload
def repartition(self, numPartitions: int, *cols: "ColumnOrName") -> "DataFrame":
    ...

@overload
def repartition(self, *cols: "ColumnOrName") -> "DataFrame":
    ...
```

Where `ColumnOrName` is a custom type defined as:
```
ColumnOrName = Union[Column, str]
```

The error I get is this:
```
error: No overloads for "repartition" match the provided arguments (reportCallIssue)
error: Argument of type "List[str] | None" cannot be assigned to parameter "cols" of type "ColumnOrName" in function "repartition" (reportArgumentType)
```

**Comment by erictraut** [comment]
If you'd like help with this, please provide a minimal, self-contained code sample that demonstrates the issue. The code above references symbols that are not defined.

**Comment by GhostlyCipher** [comment, marked as answer]
Installing https://github.com/pandas-dev/pandas-stubs fixed this issue.
